Assets/Scripts/Features/Shop/IShop.cs

```diff
@@ -0,0 +1,13 @@
+using System.Threading.Tasks;
+using Core;
+
+namespace Game
+{
+    public interface IShop : IFeature
+    {
+        Task LoadShop();
+        Task BuyProduct(string productName);
+        Task SellProduct(string productName);
+        ShopVisual Visual { get; set; }
```

Comment:
This is strange to expose the Visual. I wonder what for. Soon I will find out :)